src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/Target_ARM/ARMEmitter.cs

```diff
@@ -18,12 +18,41 @@ public ARMEmitter(NodeFactory factory, bool relocsOnly)
         public ObjectDataBuilder Builder;
         public TargetRegisterMap TargetRegister;
 
+        // mov reg, immediate
+        // reg range: 0-7
```

Comment:
Instead of adding comments specifying valid inputs, it might be more user friendly to just add an assert that enforces this.